examples/test_examples.py

```diff
@@ -556,9 +563,773 @@ def test_text_to_image_checkpointing_use_ema(self):
             pipe = DiffusionPipeline.from_pretrained(tmpdir, safety_checker=None)
             pipe(prompt, num_inference_steps=2)
 
-            # check old checkpoints do not exist
-            self.assertFalse(os.path.isdir(os.path.join(tmpdir, "checkpoint-2")))
+            self.assertEqual(
+                {x for x in os.listdir(tmpdir) if "checkpoint" in x},
+                {
+                    # no checkpoint-2 -> check old checkpoints do not exist
+                    # check new checkpoints exist
+                    "checkpoint-4",
+                    "checkpoint-6",
+                },
+            )
+
+    def test_text_to_image_checkpointing_checkpoints_total_limit(self):
```

Comment:
Every training script needs two tests: 

One: that the marginal creation of a checkpoint that would place us over the limit deletes the earliest checkpoint